connect/runtime/src/main/java/org/apache/kafka/connect/runtime/AbstractHerder.java

```diff
@@ -256,6 +256,14 @@ public ConfigInfos validateConnectorConfig(Map<String, String> connectorProps) {
             // do custom connector-specific validation
             Config config = connector.validate(connectorProps);
             ConfigDef configDef = connector.config();
+            if (null == configDef) {
```

Comment:
I guess we check for null again because the connector might override `validate`?

Reply:
What do you mean checking again? This is just checking the `ConfigDef` returned by the connector (which was only called on the previous line) and this method is called from places that wouldn't have already validated it.

Reply:
I was referring to the call to `connector.validate` two lines above. That is where the other null check in this patch in `Connector` would be applied, unless the user has overridden `validate`.

Reply:
Yes, I think it's worthwhile to check the result of `Connector.config()` just in case `Connector.validate()` is overridden and the default check there is no longer used.